File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Data.Filtering

Namespace GridViewChartFilterWeb.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function
		Public Function GridViewPartial() As ActionResult
			Return PartialView(NorthwindDataProvider.GetProducts(Nothing))
		End Function
		Public Function ChartPartial(ByVal filterExpression As String) As ActionResult
			Dim accessFilter As String = CriteriaToWhereClauseHelper.GetAccessWhere(CriteriaOperator.Parse(filterExpression))
			Return PartialView(NorthwindDataProvider.GetProducts(accessFilter))
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Data.Filtering;

namespace GridViewChartFilterWeb.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult GridViewPartial() {
            return PartialView(NorthwindDataProvider.GetProducts(null));
        }
        public ActionResult ChartPartial(string filterExpression) {
            string accessFilter = CriteriaToWhereClauseHelper.GetAccessWhere(CriteriaOperator.Parse(filterExpression));
            return PartialView(NorthwindDataProvider.GetProducts(accessFilter));
        }
    }
}
